### PR TITLE
[#402] Add install script for arduino and teensyduino

### LINKED ISSUE #402: Client side install script for arduino IDE + teensyduino
We have travis installing this automatically for every build, we should have a client-side version to make it easier on developers. Some changes will have to be made to account for the change of purpose.

### PR DESCRIPTION
**Pull request by PeterGhimself**
# Assignee Section

## Description
Added an install and setup script for arduino + teensyduino, based off the commands found in the travis config file. Some stuff had to be removed, some stuff had to be added.

- Install + setup script tested while testing for #392 
- README manual steps updated
- Included this script execution in `EnvironmentSetup.sh`

### Steps for Testing

1. Go to your arduino IDE folder (usually named something like `arduino-1.8.12`), `rm -rf <arduino_folder>`
2. Go to the root of the repo, run `./install_arduino_teensyduino.sh`
3. After the script is done, verify that it worked by navigating to your home folder where the arduion IDE folder is, `cd arduino-1.8.12`, then run `./arduino`
4. Verify that you can see in `Tools -> Board` some options with "Teensy" in it
5. (optional) If you have a teensy, try uploading a sketch to it

closes #402 

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.

- [x] Local Test Performed Successfully
- [ ] [General Integration Test](https://docs.google.com/document/d/1ug0CpA1cIzURP8DDFSvCt2CEJJSwJ6Ta6B1LG_hYk6I/edit) Performed Successfully

For Pull Requests that do not include code changes, it is not required to perform the tests above.

### CODE REVIEW

**Comment by MartensCedric** [comment]
You should append `ARDUINO_PATH` to `~/.bashrc`.

In case that its already there just do

if ARDUINO_PATH exists
do 
append ARDUINO_PATH to `~/.bashrc`
done

**Comment by PeterGhimself** [comment]
Re-tested, seems to work still

File: install_arduino_teensyduino.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ARDUINO_IDE_VERSION="1.8.12"
+TEENSYDUINO_VERSION="152"
+ARDUINO_IDE_FOLDER="$HOME/arduino-$ARDUINO_IDE_VERSION"
+ARDUINO_DOWNLOAD_URL="https://downloads.arduino.cc/arduino-$ARDUINO_IDE_VERSION-linux64.tar.xz"
+TEENSYDUINO_DOWNLOAD_URL="https://www.pjrc.com/teensy/td_$TEENSYDUINO_VERSION/TeensyduinoInstall.linux64"
+
+count=`ls -1 ./arduino-$ARDUINO_IDE_VERSION-* 2>/dev/null | wc -l`   
+
+if [ $count != 0 ]; then
+    echo "deleting left overs..."
```
Comment (MartensCedric) [inline install_arduino_teensyduino.sh:15]:
```suggestion
    echo "Deleting leftovers..."
```

File: install_arduino_teensyduino.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+# NOTE: Running this script in sudo will cause the checks for environment vars to fail
+# To complete the installation run `sudo install.sh` in the arduino IDE folder
+
+ARDUINO_IDE_VERSION="1.8.12"
+TEENSYDUINO_VERSION="152"
+ARDUINO_IDE_FOLDER="$HOME/arduino-$ARDUINO_IDE_VERSION"
+ARDUINO_DOWNLOAD_URL="https://downloads.arduino.cc/arduino-$ARDUINO_IDE_VERSION-linux64.tar.xz"
+TEENSYDUINO_DOWNLOAD_URL="https://www.pjrc.com/teensy/td_$TEENSYDUINO_VERSION/TeensyduinoInstall.linux64"
+
+count=`ls -1 ./arduino-$ARDUINO_IDE_VERSION-* 2>/dev/null | wc -l`
```
Comment (MartensCedric) [inline install_arduino_teensyduino.sh:12]: You have two `count` variables, perhaps one should be like count_of_leftover_files or changing the names to distinguish things out a bit more

**Comment by MartensCedric** [comment]
Works great!

**Pull request merged**